hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +60,115 @@
     }
 
     private long countMapWithMapStoreEnabled() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getMapStoreConfig().isEnabled()).count();
+        return mapConfigs.stream().map(MapConfig::getMapStoreConfig)
+                .filter(MapStoreConfig::isEnabled).count();
     }
 
     private long countMapWithAtleastOneQueryCache() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getQueryCacheConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getQueryCacheConfigs)
+                .filter(queryCacheConfigs -> !(queryCacheConfigs.isEmpty())).count();
     }
 
     private long countMapWithAtleastOneIndex() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getIndexConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getIndexConfigs)
+                .filter(indexConfigs -> !(indexConfigs.isEmpty())).count();
     }
 
     private long countMapWithHotRestartEnabled() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getHotRestartConfig().isEnabled()).count();
+        return mapConfigs.stream().map(MapConfig::getHotRestartConfig)
+                .filter(HotRestartConfig::isEnabled).count();
     }
 
     private long countMapWithWANReplication() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getWanReplicationRef() != null).count();
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
```

Comment:
I'm not sure if it makes sense to have this method, since it always returns the same lambda. So this can be removed and the lambda itself extracted into a constant (eg. `IS_MAP_STORE_ENABLED`).

nit: once ^this is done, the other lambda can be created as `IS_MAP_STORE_ENABLED.negate()`